File: pharmacy_app/lib/core/Models/PharmacyMedicine.dart
import 'Medicine.dart';
//import 'Pharmacy.dart';

class PharmacyMedicine {
  //int? PharmacyId;
  //Pharmacy? pharmacy;
  int? MedicineId;
  Medicine? medicine;
  int? Stock;

  PharmacyMedicine({
    //this.PharmacyId,
    //this.pharmacy,
    this.MedicineId,
    this.medicine,
    this.Stock
  });

  factory PharmacyMedicine.fromJson(Map<String,dynamic> json) {
    return PharmacyMedicine(
      //PharmacyId: json['pharmacyId'],
      //pharmacy: Pharmacy.fromJson(json['pharmacy']),
      MedicineId: json['medicineId'],
      medicine: Medicine.fromJson(json['medicine']),
      Stock: json['stock'],
    );
  }
}

File: Pharmacy_System/Pharmacy_API/Models/PharmacyMedicine.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace Pharmacy_API.Models
{
    public class PharmacyMedicine
    {
        public int PharmacyId { get; set; }
        public Pharmacy Pharmacy { get; set; }
        
        public int MedicineId { get; set; }
        public Medicine Medicine { get; set; }

        public int? Stock { get; set; }
    }
}
